frontend/src/lib/components/StrategyForm.svelte

```diff
@@ -40,28 +35,8 @@
     assetData = data;
   });
 
-  function genId() {
-    return Number($strats[$strats.length - 1]?.id || 0) + 1;
-  }
-
   async function handleSubmit() {
-    const strategy = strategyTuple(assets);
-
-    const tuple = strategy.map(BigNumber.from);
-
-    const stratData = await $treasuryContract.createStrategy(tuple);
-
-    console.log(stratData);
-
-    strats.update((st) => [
-      ...st,
-      {
-        id: genId().toString(),
-        submittedBy: stratData.from,
-        data: strategy,
-      },
-    ]);
-
+    strats.createStrategy(assets);
```

Comment:
Missing an `await` here, it will redirect while the transactino is still happening.